### PR TITLE
Use libinput transformed events instead of width_mm/height_mm

### LINKED ISSUE #456: wlroots (sway) fails to start on QEMU's VGA card
Hi

I compile my own Mesa, so it could be a Mesa issue? ...But other Wayland DEs work on this VM, and Sway does work on my Intel box

It seems that both the Wayland and DRM backend fail, due to not being able to find EGL extensions... ...however it does not say which egl extension it is trying to use, and I am using very recent Mesa

I have attached a screenshot of the error, along with some other output, since I don't have clipboard passing
swayroots

I get similar behavior with rootston

Thanks

### PR DESCRIPTION
**Pull request by SirCmpwn**
Fixes #456

### CODE REVIEW

File: backend/drm/legacy.c [hunk 1]
```diff
@@ -47,7 +47,7 @@ bool legacy_crtc_set_cursor(struct wlr_drm_backend *drm,
 
 	if (drmModeSetCursor(drm->fd, crtc->id, gbm_bo_get_handle(bo).u32,
 			plane->surf.width, plane->surf.height)) {
-		wlr_log_errno(L_ERROR, "Failed to set hardware cursor");
+		wlr_log_errno(L_DEBUG, "Failed to set hardware cursor");
```
Comment (SirCmpwn) [inline backend/drm/legacy.c]: Added an unrelated fix here

File: examples/tablet.c [hunk 2]
```diff
@@ -164,20 +162,33 @@ static void handle_pad_ring(struct tablet_pad_state *pstate,
...
+	struct compositor_state compositor = {
 		.data = &state,
 		.output_frame_cb = handle_output_frame,
 		.tool_axis_cb = handle_tool_axis,
 		.tool_proximity_cb = handle_tool_proximity,
 		.tool_button_cb = handle_tool_button,
 		.pad_button_cb = handle_pad_button,
 		.pad_ring_cb = handle_pad_ring,
+		.input_add_cb = handle_input_add,
+		0
```
Comment (martinetd) [inline examples/tablet.c]: (is that 0 needed at all? anything not specified should be zeroed out by construction)

**Comment by SirCmpwn** [comment]
https://github.com/swaywm/sway/pull/1641

File: include/wlr/types/wlr_cursor.h [hunk 3]
```diff
@@ -156,6 +156,6 @@ void wlr_cursor_map_input_to_region(struct wlr_cursor *cur,
  */
 bool wlr_cursor_absolute_to_layout_coords(struct wlr_cursor *cur,
 		struct wlr_input_device *device, double x_mm, double y_mm,
```
Comment (emersion) [inline include/wlr/types/wlr_cursor.h]: `x_mm` → `x`, `y_mm` → `y`

**Comment by emersion** [comment]
Input is fucked up for me in the Wayland backend. I can't click anything.

Also make sure to try with multiple transformed outputs.

**Comment by martinetd** [comment]
> Input is fucked up for me in the Wayland backend. I can't click anything.
> 
> Also make sure to try with multiple transformed outputs.

works for me with no transform (or just scale or 180 transform), but 90/270 rotations were broken before this commit, WL backend or not.

I also just got a segv on wl backend shutdown, will open an issue to not forget that one...

**Comment by SirCmpwn** [comment]
Thanks!

**Comment by emersion** [comment]
Wait, how does this PR fixes #456?

It fixes #639 though.

**Comment by SirCmpwn** [comment]
Read the whole discussion on #456.

**Pull request merged**